aten/src/ATen/native/transformers/cuda/sdp_utils.cpp

```diff
@@ -173,6 +173,23 @@ bool check_flash_attention_hardware_support(sdp_params const& params, bool debug
   using sm80 = SMVersion<8, 0>;
   using sm90 = SMVersion<9, 0>;
   auto dprops = at::cuda::getCurrentDeviceProperties();
+#if USE_ROCM
+  constexpr std::string_view mi200 = "gfx90a:sramecc+:xnack-";
+  const char* real_arch = dprops->gcnArchName;
+  const char* over_arch = std::getenv("PYTORCH_DEBUG_FLASH_ATTENTION_GCN_ARCH_OVERRIDE");
```

Comment:
can you explain this, it feels like reading an env flag at every call to SDPA is not the right thing to do.

Reply:
Oh this is a debugging mechanism to check tests are properly disabled even without non-MI200 accesses, but I agree we should access env vars at every SDPA call and I'll cache the result in a later patch.

Reply:
Fixed in 269683336c245a3c248d4eaf095020f42e4f6d3c and c8df600608f0690cd38eb39d95c6de60b9eff586
Now it's guarded by `c10::call_once`